dev-init: attach-smoke fails with "missing apiVersion v1beta1" when running daemon predates #641 (kuke daemon reset does not replace stale daemon binary)

## Repro

On a host where the running \`kukeond\` was built before #641/#645 (the change that switched the per-container metadata renderer from sbsh TerminalDoc to kukeon ContainerDoc — commit 3d59273, 2026-05-21 17:55), bring the worktree to current \`main\` (verified at commit 3130818 today, 2026-05-25) and run:

\`\`\`bash
make dev-init
\`\`\`

The first phases pass (build, daemon reset, image stage, daemon-parity check on \`kuke get realms\`). The script then enters the \`==> kuke attach smoke against a kuketty-wrapped cell\` section, creates a \`cattach\` cell with an attachable busybox container, waits 20 s for the kuketty socket to bind (which succeeds — the script does not trip the "kuketty socket not bound" timeout), and then fails the next check:

\`\`\`
rendered metadata at /opt/kukeon/data/dev-init-attach/ds/dks/cattach/work/kuketty-metadata.json missing apiVersion v1beta1
make: *** [Makefile:209: dev-init] Error 1
\`\`\`

## Observed

Inspecting \`kuketty-metadata.json\` on this host before the script's auto-purge fires (re-created the smoke cell manually post-failure to capture the on-disk content):

\`\`\`json
{
  "apiVersion": "sbsh/v1beta1",
  "kind": "Terminal",
  "metadata": {
    "name": "c",
    "labels": {
      "kukeon.io/cell": "cattach",
      ...
    }
  },
  "spec": {
    "id": "023063f1",
    "kind": 0,
    "name": "gallant_amonhen",
    ...
    "runPath": "/run/kukeon/tty",
    "captureFile": "/run/kukeon/tty/capture",
    "logFile": "",
    "socketIO": "/run/kukeon/tty/socket",
    ...
  }
}
\`\`\`

The on-disk file is a pre-#641 sbsh \`TerminalDoc\` (\`apiVersion: sbsh/v1beta1\`, \`kind: Terminal\`), not the post-#641 kukeon \`ContainerDoc\` (\`apiVersion: v1beta1\`, \`kind: Container\`) the daemon's current \`writeKukettyDoc\` (\`internal/controller/runner/attachable.go:214 writeKukettyDoc\`) renders. The script's grep at \`scripts/dev-init.sh:311\` is correct against current-code expectations — the file content is the divergence.

The running daemon (\`ps -o pid,lstart\` on the \`/bin/kukeond serve\` process) was started 2026-05-21 06:51:15, ~11 hours **before** #641 merged. The kukeond cell on disk (\`/opt/kukeon/data/kuke-system/kukeon/kukeon/kukeond/\`) was created at the same timestamp; \`make dev-init\`'s "Reset prior kukeond cell" phase (\`scripts/dev-init.sh:192 sudo ./kuke daemon reset\`) ran today, but the daemon process PID did not change across the run and the running daemon's binary is still the pre-#641 build. The \`kuke build\` step did produce a fresh \`docker.io/library/kukeon-local:dev\` image into the kuke-system realm's containerd namespace, but \`kuke init --kukeond-image\` apparently kept the existing kukeond cell in place (consistent with the script's own log output:

\`\`\`
- cell "kukeond": already existed
- cell cgroup: already existed
- cell root container: already existed
- cell containers: started
\`\`\`

— \`started\` covers "already running," not "replaced with new image").

Confirmed cross-branch: the failure reproduces identically on \`origin/main\` (3130818) under an isolated worktree on the same host, so it's not specific to any in-flight PR.

## Root cause

Not yet pinpointed. Candidate shortlist for the next picker:

1. **\`kuke daemon reset\` is not destructive enough to force a binary swap on the next \`kuke init\`.** The CLAUDE.md description ("stops + deletes the prior \`kukeond\` cell and clears \`/run/kukeon/kukeond.{sock,pid}\`") suggests the cell should be deleted, but the script's subsequent \`kuke init\` log line "cell "kukeond": already existed" indicates the cell record survived. If reset is supposed to delete it, that's a reset bug. \`internal/controller/runner/\` (the reset path) and \`cmd/kuke/daemon/reset/reset.go\` are the candidates.
2. **\`kuke init --kukeond-image\` is idempotent against the running cell** and treats an already-Ready kukeond as up-to-date even when \`--kukeond-image\` resolves to a digest the running container does not have. The fix would be to compare the running container's image digest against \`--kukeond-image\` and replace on mismatch. \`internal/controller/runner/\` (the init/provision path) and \`cmd/kuke/init/init.go\` are the candidates.
3. **\`make dev-init\` flow gap** — the script trusts \`kuke daemon reset\` + \`kuke init\` to converge on the fresh image and never verifies the running daemon's version against the just-built image. \`scripts/dev-init.sh\` could add a post-init "running kukeond image digest must match \`\${KUKEOND_IMAGE_REF}\`" check that fails-loud before the smoke step, regardless of whether the underlying cause is in reset or init.

The user-facing failure under each candidate is identical, so all three would be fixed by adding the post-init image-digest check at the bare minimum, and the root cause then surfaces as either "reset didn't delete the cell" or "init didn't replace the image."

## Suggested fix

Two-layer fix:

1. \`scripts/dev-init.sh\` — after \`kuke init\`, query the running kukeond container's image digest (e.g. \`sudo ctr -n kuke-system.kukeon.io container info kukeon_kukeon_kukeond_kukeond | jq -r '.Image'\`) and compare against the just-staged \`\${KUKEOND_IMAGE_REF}\` digest from \`sudo ctr -n kuke-system.kukeon.io images ls\`. Fail-loud with "running kukeond image \`<old>\` does not match staged \`<new>\` — try \`kuke daemon reset --purge-system\` to force a clean reprovision" before the smoke phase runs. This catches the gap regardless of where the underlying bug is.
2. \`internal/controller/runner/\` or \`cmd/kuke/daemon/reset/reset.go\` — investigate whether the reset path is actually deleting the kukeond cell record (the \`--purge-system\` flag suggests it currently keeps user-realm data; the \`kukeond\` cell itself should be unconditionally deleted on reset). If reset is correct, then \`kuke init\` needs a "image-digest mismatch → recreate" branch on the existing-cell path.

Memory hint: the [Init/bootstrap self-sufficiency gaps are bugs](../../feedback_self_sufficiency_gaps_are_bugs.md) feedback applies — the operator should not need to manually \`kuke uninstall\` + reinstall after upgrading the source tree across a daemon-affecting change. \`make dev-init\` is documented as "the canonical re-bootstrap loop" (project CLAUDE.md) and the user contract is "re-running on a healthy host produces a clean re-bootstrap."

--- a/scripts/dev-init.sh
+++ b/scripts/dev-init.sh
@@ -207,6 +207,80 @@ sudo --preserve-env=KUKEON_HOST ./kuke build --build-arg VERSION=v0.0.0-dev \
 step "Run kuke init with --kukeond-image ${KUKEOND_IMAGE_REF}"
 sudo --preserve-env=KUKEON_HOST,KUKEOND_SOCKET ./kuke init --kukeond-image "${KUKEOND_IMAGE_REF}"
 
+# Fail-loud when the running kukeond container is anchored on a stale
+# image (issue #857). The bug captured there is that `kuke daemon reset`
+# + `kuke init --kukeond-image` does not refresh the kukeond cell when
+# the image tag is reused (init's bootstrapCell takes the existing-cell
+# path on a leftover record and never compares the desired image
+# against the running container). The user-visible failure surfaces
+# downstream as an opaque pre-#641 schema mismatch in the attach smoke;
+# this check turns that into a fail-loud at the place the divergence
+# was introduced.
+#
+# Compare the *snapshot chain digest*, not the image ref name: a
+# container's `.Image` field is the ref string set at create time
+# (e.g. `docker.io/library/kukeon-local:dev`) and is unchanged when
+# `kuke build` overwrites the same tag with a new manifest. The
+# canonical drift signal is the chain digest stored on the container's
+# snapshot at unpack time, derived from the image config blob's
+# `containerd.io/gc.ref.snapshot.overlayfs` label. The two should be
+# byte-identical on a healthy `make dev-init`; a mismatch means the
+# running container is anchored on an older image than the one we just
+# built.
+#
+# Stay grep/sed-only — scripts in this repo intentionally avoid a jq
+# dependency (see scripts/install.sh's "no jq dependency" guarantee).
+step "Verify running kukeond container is anchored on the just-built image"
+KUKEOND_NS="kuke-system.kukeon.io"
+KUKEOND_CTR_ID="kukeon_kukeon_kukeond_kukeond"
+
+ctr_chain="$(sudo ctr -n "${KUKEOND_NS}" snapshots info "${KUKEOND_CTR_ID}" 2>/dev/null \
+    | grep -oE '"Parent":[[:space:]]*"sha256:[a-f0-9]+"' \
+    | sed -E 's/.*"(sha256:[a-f0-9]+)"/\1/')"
+if [ -z "${ctr_chain}" ]; then
+    printf 'unable to resolve snapshot chain for running kukeond container %s/%s\n' \
+        "${KUKEOND_NS}" "${KUKEOND_CTR_ID}" >&2
+    exit 1
+fi
+
+manifest_digest="$(sudo ctr -n "${KUKEOND_NS}" images ls 2>/dev/null \
+    | awk -v r="${KUKEOND_IMAGE_REF}" 'NR>1 && $1==r {print $3; exit}')"
+if [ -z "${manifest_digest}" ]; then
+    printf 'image %s not found in containerd namespace %s after kuke build\n' \
+        "${KUKEOND_IMAGE_REF}" "${KUKEOND_NS}" >&2
+    exit 1
+fi
+
+config_digest="$(sudo ctr -n "${KUKEOND_NS}" content get "${manifest_digest}" 2>/dev/null \
+    | grep -oE '"digest":[[:space:]]*"sha256:[a-f0-9]+"' \
+    | head -1 \
+    | sed -E 's/.*"(sha256:[a-f0-9]+)"/\1/')"
+if [ -z "${config_digest}" ]; then
+    printf 'unable to resolve config digest from manifest %s\n' "${manifest_digest}" >&2
+    exit 1
+fi
+
+img_chain="$(sudo ctr -n "${KUKEOND_NS}" content ls 2>/dev/null \
+    | awk -v d="${config_digest}" '$1==d' \
+    | grep -oE 'containerd\.io/gc\.ref\.snapshot\.overlayfs=sha256:[a-f0-9]+' \
+    | head -1 \
+    | cut -d= -f2)"
+if [ -z "${img_chain}" ]; then
+    printf 'unable to resolve image chain digest from config blob %s\n' "${config_digest}" >&2
+    exit 1
+fi
+
+if [ "${ctr_chain}" != "${img_chain}" ]; then
+    printf 'running kukeond container is anchored on a stale image (#857)\n' >&2
+    printf '  running container snapshot chain: %s\n' "${ctr_chain}" >&2
+    printf '  just-built image chain:           %s\n' "${img_chain}" >&2
+    printf '  manifest %s did not propagate to the kukeond cell —\n' "${manifest_digest}" >&2
+    printf '  `kuke daemon reset` + `kuke init --kukeond-image` did not refresh the container.\n' >&2
+    printf '  Try `sudo ./kuke daemon reset --purge-system` then re-run `make dev-init`.\n' >&2
+    exit 1
+fi
+echo "running kukeond container chain ${ctr_chain} matches just-built ${KUKEOND_IMAGE_REF}"
+
 step "Daemon parity check (both must show identical output)"
 sudo --preserve-env=KUKEON_HOST ./kuke get realms
 sudo ./kuke get realms --no-daemon

